common/functional_tester.py

```diff
@@ -0,0 +1,19 @@
+import fault
+
+
+class FunctionalTester(fault.Tester):
```

Comment:
I like that idea, makes sense. I'll implement it. Also, maybe this should move to fault as well? I'm not quite sure since at the moment it's pretty tied to the way we defined our functional model abstraction, but if we generalize that notion (basically, formalize the interface for general hardware circuits) this could be useful to any users of Fault.

Reply:
Yeah agreed. What if we moved the functional model interfaces to fault as well? Could be a forcing function to solidify those interfaces.